src/main/java/Duke.java

```diff
@@ -1,10 +1,297 @@
+import java.util.Scanner;
+
 public class Duke {
+
+    private static final int MAX_TASK = 100;
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        printHello();
+        runProgram();
+        printBye();
+    }
+
+    private static void runProgram() {
+        Task[] taskList = new Task[MAX_TASK];
+        Scanner in = new Scanner(System.in);
+
+        //Loop to receive response.
+        while (true){
+
+            String input = in.nextLine();
+
+            // BYE command
+            if(input.equalsIgnoreCase("bye")){
+                return;
+            }
+
+            // LIST command
+            if(input.equalsIgnoreCase("list")){
+                runList(taskList);
+            }
+
+            //HELP COMMAND
+            else if(input.equalsIgnoreCase("help")){
+                printHelp();
+            }
+
+            // DONE command
+            else if(startsWith(input, "done")){
+                runDone(taskList, input);
+            }
+
+            // TO-DO COMMAND
+            else if(startsWith(input, "todo")){
+                runTodo(taskList, parseJob(input, ""));
+            }
+
+            // DEADLINE COMMAND
+            else if(startsWith(input, "deadline")){
+                runDeadline(taskList, parseJob(input, "/by"), parseDate(input, "/by"));
+            }
+
+            // EVENTS COMMAND
+            else if(startsWith(input, "event")){
+                runEvent(taskList, parseJob(input, "/at"), parseDate(input, "/at"));
+            }
+
+            // ADD command
+            else {
+                runAdd(taskList, input);
+            }
+        }
+
+
+    }
+
+    /**
+     * COMMAND RUNNER METHODS
+     * */
+
+    private static void runAdd(Task[] taskList, String input) {
+        // stores user command as job
+        Task newTask = new Task(input);
+
+        taskList[Task.taskCount] = newTask;
+        Task.taskCount++;
+
+        printTaskAdded(newTask);
+    }
+
+    private static void runDone(Task[] taskList, String input) {
+        String[] word = input.split(" ");
+
+        // check invalid input
+        if(word.length != 2){
+            printInvalidInputWarning();
+            return;
+        }
+
+        int jobNumber = Integer.parseInt(word[1]) - 1;
+
+        if(jobNumber < Task.taskCount && jobNumber >= 0){
+            markJobAsDone(taskList[jobNumber]);
+        }
+        else if(Task.taskCount == 0){
+            printNoTaskWarning();
+        }
+        else{
+            printInvalidTaskWarning(jobNumber);
+        }
+    }
+
+    private static void runList(Task[] taskList) {
+        int numbering = 1;
+
+        // error handling - no jobs
```

Comment:
Overall, all comments appear in the right position of your code. Good following of the standard.